updating pref information on firefox support for web components features

chrisdavidmills opened the pull request:
I found out recently that Firefox prefs for web components features have changed. you now only need:

* dom.webcomponents.customelements.enabled for custom elements features.
* dom.webcomponents.shadowdom.enabled for shadow DOM features.

The old dom.webcomponents.enabled has now been completely removed.

I also added in support information for the "is" global attribute, seen as it is closely related.

Elchi3 commented:
The global `is`  attribute will get added in another PR already and I think the way the Firefox support is indicated there records the whole support history here. What do you think about it: https://github.com/mdn/browser-compat-data/pull/1282/files#diff-2a52c9b7d3948297f271ac5ef0a10070R546 ?

chrisdavidmills commented:
That PR info looks good. I have therefore removed the "is" information from this one.

Elchi3 commented:
Over in the other PR the Firefox situation is described like this:

```json
   "firefox": [
              {
                "version_added": "59",
                "flags": [
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.customelements.enabled",
                    "value_to_set": "true"
                  }
                ]
              },
              {
                "version_added": true,
                "version_removed": "59",
                "flags": [
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.enabled",
                    "value_to_set": "true"
                  },
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.customelements.enabled",
                    "value_to_set": "true"
                  }
                ]
              }
],
```
Does this describe the flag name changes better and would this apply to (all / some of?) the features you're changing as part of this PR?

chrisdavidmills commented:
I guess this more accurate, strictly speaking. I'll update the others in this PR.

Elchi3 commented:
Looks good to me. Maybe @ExE-Boss wants to give this a review, too. 
And I think this PR is also conflicting with https://github.com/mdn/browser-compat-data/pull/1253

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -21,36 +21,62 @@
+            },
+            {
+              "version_added": true,
+              "version_removed": "59",
```
This feature was unavailable before Firefox 59.

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -21,36 +21,62 @@
+            },
+            {
+              "version_added": true,
+              "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -94,36 +120,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -94,36 +120,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -168,36 +220,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -168,36 +220,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -242,36 +320,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on api/ShadowRoot.json:
```diff
@@ -242,36 +320,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on css/selectors/slotted.json:
```diff
@@ -21,36 +21,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

ExE-Boss commented on css/selectors/slotted.json:
```diff
@@ -21,36 +21,62 @@
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
ditto

chrisdavidmills commented:
OK, so for my latest trick, I've fixed slotted as per @ExE-Boss 's review. I've also reverted the ShadowRoot data back to what it currently is in master.

My thinking is that if we merge this PR, and then #1253 , all should work out correct? @Elchi3 ?
